membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -361,58 +351,35 @@ case class Catalog(
   lazy val productMap: Map[ProductRatePlanChargeId, Benefit] =
     map.values.flatMap(p => p.benefits).toMap
 
-  lazy val paid: Seq[CatalogPlan.Paid] =
-    supporter.plans ++ partner.plans ++ patron.plans ++ allSubs.flatten
-
-  lazy val allSubs: List[List[CatalogPlan.Paid]] =
-    List(digipack.plans, supporterPlus.plans, voucher.list.toList, digitalVoucher.list.toList, delivery.list.toList) ++ weekly.plans
 }
 
 /** A higher level representation of a number of Zuora rate plan charges
   */
 sealed trait ChargeList {
   def benefits: NonEmptyList[Benefit]
-  def currencies: Set[Currency]
-}
-
-sealed trait FreeChargeList extends ChargeList {
-  def currencies: Set[Currency]
-}
-
-sealed trait PaidChargeList extends ChargeList {
   def gbpPrice = price.getPrice(GBP).getOrElse(throw new Exception("No GBP price"))
   def currencies = price.currencies
   def billingPeriod: BillingPeriod
   def price: PricingSummary
   def subRatePlanChargeId: SubscriptionRatePlanChargeId
 }
 
-/** Generic version of single free / paid charge This is to allow exhaustive matches on tier in membership
-  * i.e. the common ancestor type of Friend / Supporter will be Plan[ChargeList with SingleBenefit[MemberTier]] as opposed to just Plan[ChargeList]
-  * which isn't typed to only contain member tiers
+/** Same as above but we must have exactly one charge, giving us exactly one benefit This is used for supporter, partner, patron and digital pack subs
   */
-sealed trait SingleBenefit[+B <: Benefit] {
-  def benefit: B
-}
```

Comment:
SingleBenefit was not used any more.  We can probably get rid of benefit for subscriptions at runtime completely, although it's still needed for catalog (unless we pull in a copy of the Map from product to product rate plan id, so that we deserialise based on IDs rather than shape)